### PR TITLE
Refactor code to bring staffing logic to StaffBot

### PR DESCRIPTION
**Pull request by joshblum**

### CODE REVIEW

**Comment by joshblum** [comment]
depends on #231

File: orchestra/communication/staffing.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+from annoying.functions import get_object_or_None
+from django.db import transaction
+
+from orchestra.models import StaffingRequest
+from orchestra.models import StaffingResponse
+
+
+@transaction.atomic
+def handle_staffing_response(worker, pk, is_available):
```
Comment (marcua) [inline orchestra/communication/staffing.py:9]: staffing_request_id

File: orchestra/tests/test_task_lifecycle.py [hunk 2]
```diff
@@ -36,34 +36,35 @@ class BasicTaskLifeCycleTestCase(OrchestraTestCase):
     """
     Test modular functions in the task_lifecycle
     """
+
     def setUp(self):  # noqa
         super().setUp()
         setup_models(self)
 
-    def test_worker_certified_for_task(self):
+    def testis_worker_certified_for_task(self):
```
Comment (marcua) [inline orchestra/tests/test_task_lifecycle.py:44]: test_is

File: orchestra/communication/staffing.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    """
```
Comment (marcua) [inline orchestra/communication/staffing.py:10]: empty docstring

File: orchestra/bots/staffbot.py [hunk 4]
```diff
@@ -74,3 +82,57 @@ def __init__(self, **kwargs):
     def dispatch(self, data):
         data = self.validate(data)
         return {'text': data.get('text')}
+
+    def _send_task_to_workers(self, task, required_role):
+        # get all the workers that are certified to complete the task.
+        workers = Worker.objects.all()
```
Comment (marcua) [inline orchestra/bots/staffbot.py:88]: now that workflows/steps are in the DB, I think we can push is_worker_certified_for_task and check_worker_allowed_new_assignment into the DB as filters so we don't loop over all workers

**Pull request merged**